File: jobapplicationtracker/jdbc_view_applications.java
import java.sql.*;

public class jdbc_view_applications {
    // The main program that views applications
    public static void main(String[] args) throws SQLException {
        String Username = "ajm057"; // Change to your own username
        String mysqlPassword = "de2ki3Ah"; // Change to your own mysql Password

        // Connect to the database
        jdbc_db myDB = new jdbc_db();
        myDB.connect(Username, mysqlPassword);
        myDB.initDatabase();

        // Sets the selected option in dropdown to args[0] and if nothing is selected
        // say error
        String selectedOption = null;
        if (args.length > 0) {
            selectedOption = args[0];
        } else {
            System.out.println("No option selected");
            System.exit(1);
        }

        String idValue;
        idValue = args[1];
        // Switch statment to call certain queries based on selected/inputted data
        String query = "";
        switch (selectedOption) {
            case "CSCE":
                query = "SELECT S.STUDENT_NAME, J.COMPANY_NAME, J.SALARY, S.MAJOR FROM APPLICATIONS A JOIN STUDENTS S ON A.STUDENT_ID = S.STUDENT_ID JOIN JOBS J ON A.JOB_ID = J.JOB_ID WHERE S.MAJOR = 'CSCE';";
                break;
            case "BIOL":
                query = "SELECT S.STUDENT_NAME, J.COMPANY_NAME, J.SALARY, S.MAJOR FROM APPLICATIONS A JOIN STUDENTS S ON A.STUDENT_ID = S.STUDENT_ID JOIN JOBS J ON A.JOB_ID = J.JOB_ID WHERE S.MAJOR = 'BIOL';";
                break;
            case "MATH":
                query = "SELECT S.STUDENT_NAME, J.COMPANY_NAME, J.SALARY, S.MAJOR FROM APPLICATIONS A JOIN STUDENTS S ON A.STUDENT_ID = S.STUDENT_ID JOIN JOBS J ON A.JOB_ID = J.JOB_ID WHERE S.MAJOR = 'MATH';";
                break;
            case "STUDENT":
                // If a dropdown was selected and an id was inputted
                if (args.length > 1) {
                    query = "SELECT S.STUDENT_NAME, J.COMPANY_NAME, J.SALARY, S.MAJOR FROM APPLICATIONS A JOIN STUDENTS S ON A.STUDENT_ID = S.STUDENT_ID JOIN JOBS J ON A.JOB_ID = J.JOB_ID WHERE S.STUDENT_ID = "
                            + idValue;
                } else {
                    System.out.println("No student ID provided");
                    System.exit(1);
                }
                break;
            case "JOB":
                // If a dropdown was selected and an id was inputted
                if (args.length > 1) {
                    query = "SELECT S.STUDENT_NAME, J.COMPANY_NAME, J.SALARY, S.MAJOR FROM APPLICATIONS A JOIN STUDENTS S ON A.STUDENT_ID = S.STUDENT_ID JOIN JOBS J ON A.JOB_ID = J.JOB_ID WHERE J.JOB_ID = "
                            + idValue;
                } else {
                    System.out.println("No job ID provided");
                    System.exit(1);
                }
                break;
            case "ALL":
                query = "SELECT S.STUDENT_NAME, J.COMPANY_NAME, J.SALARY, S.MAJOR FROM APPLICATIONS A JOIN STUDENTS S ON A.STUDENT_ID = S.STUDENT_ID JOIN JOBS J ON A.JOB_ID = J.JOB_ID;";
                break;
            default:
                System.out.println("Invalid option");
                System.exit(1);
        }

        StringBuilder builder = new StringBuilder();
        builder.append("<br><br><br> Table STUDENTS results:" + myDB.query(query));
        System.out.println(builder.toString());

        myDB.disConnect();
    }
}

File: jobapplicationtracker/jdbc_view_applications.php
<html>
<body style="background-color:navajowhite;">
    <h3>Select an option to view all applications for the selected option</h3>
    <form action="jdbc_view_applications.php" method="post"> 
        <select id="method" name="method">
            <option value="CSCE">CSCE</option>
            <option value="BIOL">BIOL</option>
            <option value="MATH">MATH</option>
            <option value="STUDENT">STUDENTID</option>
            <option value="JOB">JOBID</option>
            <option value="ALL">ALL</option>
        </select>
        <input type="text" id="idValue" name="idValue" placeholder="Enter ID Value" style="display:none;">
        <br><br>
        <input type="submit" value="Submit">
    </form>
    <script>
        // Add event listener to the dropdown
        document.getElementById("method").addEventListener("change", function() {
            // Get whatever is selected
            var selectedOption = this.value;
            // Receive input
            var idValueInput = document.getElementById("idValue");
            // If student option or job option is selected display the id text box else hide it
            if (selectedOption === "STUDENT" || selectedOption === "JOB") {
                idValueInput.style.display = "block";
            } else {
                idValueInput.style.display = "none";
            }
        });
    </script>
</body>
<a href="http://www.csce.uark.edu/~ajm057/Assignment_5/homepage.html"><button style="font-size: 24px; padding: 5px 15px;">HOME PAGE</button></a>
</html>

<?php
if (isset($_POST['method'])) 
{
    $method = $_POST['method'];
    // Is ID value set to something else set it to blank
    $idValue = isset($_POST['idValue']) ? $_POST['idValue'] : '';

    // Replace ' ' with '\ ' in the strings so they are treated as single command line args
    $method = escapeshellarg($method);
    $idValue = escapeshellarg($idValue);

    $command = 'java -cp .:mysql-connector-java-5.1.40-bin.jar jdbc_view_applications ' . $method . ' ' . $idValue;

    // Remove dangerous characters from command to protect web server
    $escaped_command = escapeshellcmd($command);
    echo "<p>command: $command <p>"; 
    // Run jdbc_view_students.exe
    system($escaped_command);           
}
?>